src/metabase/db/custom_migrations.clj

```diff
@@ -105,6 +105,38 @@
   (when s
     (.toLowerCase (str s) Locale/US)))
 
+(defn json-in
+  "Default in function for columns given a Toucan type `:json`. Serializes object as JSON."
+  [obj]
+  (if (string? obj)
+    obj
+    (json/generate-string obj)))
+
+(defn- json-out [s keywordize-keys?]
+  (if (string? s)
+    (try
+      (json/parse-string s keywordize-keys?)
+      (catch Throwable e
+        (log/error e "Error parsing JSON")
+        s))
+    s))
+
+(def ^:private encrypted-json-in
+  (comp encryption/maybe-encrypt json-in))
+
+(defn- encrypted-json-out
+  [v]
+  (let [decrypted (encryption/maybe-decrypt v)]
+    (try
+      (json/parse-string decrypted true)
+      (catch Throwable e
+        (if (or (encryption/possibly-encrypted-string? decrypted)
+                (encryption/possibly-encrypted-bytes? decrypted))
+          (log/error e "Could not decrypt encrypted field! Have you forgot to set MB_ENCRYPTION_SECRET_KEY?")
```

Comment:
Doesn't it make more sense to just fetch and check the value of the secret key?

Reply:
Yea, we can probably get a better error message here by doing that. 

But we have [tests](https://github.com/metabase/metabase/blob/d7d7384c1443f5c4cc29461671971beb0ea5eae2/test/metabase/models/interface_test.clj#L149) for it, and I don't want to change some existing behavior as part of an escalation fix.